### DISCUSSION TITLE
How to remove signal event listener

### DISCUSSION THREAD

**Discussion by khidirbekov** [reduxjs/redux-toolkit#3308] [Q&A] [answered]
Hi! In the documentation of createAsyncThunk example with abort axios request. 
<img width="974" alt="image" src="https://user-images.githubusercontent.com/39237435/228481214-e79596f3-bd47-4ed1-8b15-1e92ef900047.png">

And there is 
```
signal.addEventListener('abort', () => {
      source.cancel()
    })
```

But I don't see where is removeEventListener. Am I correct that after call source.cancel abort listener is removing? Or I should to do something like this
```
createAsyncThunk(
        'LIST/fetchList',
        async (
            params,
            { signal },
        ) => {
            const source = axios.CancelToken.source();
            const cancelRequest = () => source.cancel();

            signal.addEventListener('abort', cancelRequest);

            try {
                const { data } = await apiMethod(
                   params,
                    {
                        cancelToken: source.token,
                    },
                );

                return data;
            } catch (error) {
                throw error;
            } finally {
                signal.removeEventListener('abort', cancelRequest);
            }
        },
    );
```

Or there is any way?

**Comment by phryneas** [comment, marked as answer]
I don't think you really need to remove that - the signal will be thrown away after the request, and only be used for that one individual request.

**Comment by khidirbekov** [comment]
Thanks u very much for the answer
